cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/TriggerHandler.java

```diff
@@ -143,11 +127,14 @@ private void injectInformationForBibsysHostedCustomer(UserAttributes userAttribu
         userAttributes.setAffiliation(extractAffiliationFromHostedUSer(userAttributes.getHostedAffiliation()));
     }
 
-    private UserDetails createUserDetails(UserAttributes userAttributes) {
-        return Optional.ofNullable(userAttributes.getOrgNumber())
-                .flatMap(orgNum -> mapOrgNumberToCustomer(removeCountryPrefix(orgNum)))
-                .map(customer -> new UserDetails(userAttributes, customer))
-                .orElse(new UserDetails(userAttributes));
+    private UserDetails createUserDetails(Event event) {
+        return Optional.of(event)
+            .map(Event::getRequest)
+            .map(Request::getUserAttributes)
+            .map(UserAttributes::getOrgNumber)
+            .flatMap(orgNum -> mapOrgNumberToCustomer(removeCountryPrefix(orgNum)))
```

Comment:
Minor but seeing it now maybe we should remove country prefix in its own map() call